pkg/operator/vttablet/pod.go

```diff
@@ -61,6 +61,8 @@ func NewPod(key client.ObjectKey, spec *Spec) *corev1.Pod {
 func UpdatePodInPlace(obj *corev1.Pod, spec *Spec) {
 	// Update labels and annotations, but ignore existing ones we don't set.
 	update.Labels(&obj.Labels, spec.Labels)
+
+	updatePodAnnotations(obj, spec)
```

Comment:
I didn't think we talked about changing the behavior of other annotations besides those set through VitessShardSpec.Annotations. The use case we have for VitessShardSpec.Annotations doesn't apply here, so we'd have to separately justify this change and verify that it's safe.

Reply:
Alright, that's a good point. I can revert the pod annotation changes.

Reply:
Done.